lib/timescale/acts_as_hypertable.rb

```diff
@@ -0,0 +1,101 @@
+# frozen_string_literal: true
+
+module Timescale
+  # If you want your model to hook into its underlying hypertable
+  # as well as have access to TimescaleDB specific data, methods, and more,
+  # specify this macro in your model.
+  #
+  # @note Your model's table needs to have already been converted to a hypertable
+  # via the TimescaleDB `create_hypertable` function for this to work.
+  #
+  # @see https://docs.timescale.com/api/latest/hypertable/create_hypertable/ for
+  #   how to use the SQL `create_hypertable` function.
+  # @see Timescale::MigrationHelpers#create_table for how to create a new hypertable
+  # via a Rails migration utilizing the standard `create_table` method.
+  #
+  # @example Enabling the macro on your model
+  #   class Event < ActiveRecord::Base
+  #     acts_as_hypertable
+  #   end
+  #
+  # @see Timescale::ActsAsHypertable::ClassMethods#acts_as_hypertable
+  #   for configuration options
+  module ActsAsHypertable
+    extend ActiveSupport::Concern
+
+    module ClassMethods
+      # == Configuration options
+      #
+      # @param [Hash] options The options to initialize your macro with.
+      # @option options [Symbol] :time_column The name of the column in your
+      #   model's table containing time values. The name provided should be
+      #   the same name as the `time_column_name` you passed to the
+      #   TimescaleDB `create_hypertable` function.
+      #
+      # @example Enabling the macro on your model with options
+      #     class Event < ActiveRecord::Base
+      #       acts_as_hypertable time_column: :timestamp
+      #     end
+      #
+      def acts_as_hypertable(options = {})
+        return if included_modules.include?(Timescale::ActsAsHypertable::HyperableInstanceMethods)
```

Comment:
As the module is empty, we can safely remove it.

```suggestion
```

Reply:
I'm okay with removing the empty `Timescale::ActsAsHypertable::HyperableInstanceMethods` module, however, we should retain this guard clause. Just means we'll need a different conditional to check against.

For context, the purpose of the guard clause is to prevent multiple macro calls. If a developer calls this macro twice (without noticing), the second macro call will override the options set in the first call.

As an example, if a developer is not attentive to their Model and has this:

```ruby
class Event < ApplicationRecord
  # ...

  acts_as_hypertable time_column: :timestamp
  # Event.hypertable_options => {time_column: :timestamp}

  #
  # chunk of code which potentially obscured the first call
  #

  acts_as_hypertable
  # Event.hypertable_options => {time_column: :created_at}
end
```

The second call is going to redefine the `hypertable_options` [defined here](https://github.com/jonatas/timescale/pull/2/files#diff-1da9e137201481b9a6ec29702285eee685e074c6625d077b50cafc9d05652bfcR48).